Backbone.CollectionBinder.js

```diff
@@ -160,18 +160,94 @@
         },
 
         sortRootEls: function(){
-            this._collection.each(function(model, modelIndex){
+            // sorting an empty collection or a collection with a single element won't change anything
+            if (this._collection.length < 2) return;
+
+            // Algorithm: one by one take elements and put to the correct position in DOM
+            // Computational complexity: O(n^2)
+            // DOM operations:
+            // - $.children() calls: 1
+            // - $.before/after calls: n
+
+
+            // -----
+            // prepare data for sorting
+
+            var children = $(this._elManagerFactory._getParentEl()).children();
+
+            var order = 0;
+
+            var getElData = function (model) {
                 var modelElManager = this.getManagerForModel(model);
-                if(modelElManager){
-                    var modelEl = modelElManager.getEl();
-                    var currentRootEls = $(this._elManagerFactory._getParentEl()).children();
+                if (!modelElManager) return;
 
-                    if(currentRootEls[modelIndex] !== modelEl[0]){
-                        modelEl.detach();
-                        modelEl.insertBefore(currentRootEls[modelIndex]);
-                    }
+                var modelEl = modelElManager.getEl();
+                if (!modelEl) return;
+
+                return {
+                    el: modelEl,
+                    elIndex: children.index(modelEl),
+                    order: order++
                 }
-            }, this);
+            };
+
+            var elements = _.chain(this._collection.models)
```

Comment:
You could also use `this._collection.chain()`, which I think reads better and also allows for implementation changes on where models are stored in a collection in the future.